cilium/bpf_metadata.cc

```diff
@@ -295,52 +353,16 @@ bool Config::getMetadata(Network::ConnectionSocket& socket) {
 
       east_west_l7_lb = true;
 
-      // Keep the original source address for the matching IP version, create a new source IP for
-      // the other version (with the same source port number) in case an upstream of a different
-      // IP version is chosen.
-      const auto& ips = policy->getEndpointIPs();
-      switch (sip->version()) {
-      case Network::Address::IpVersion::v4: {
-        ipv4_source_address = src_address;
-        if (ips.ipv6_) {
-          sockaddr_in6 sa6 = *reinterpret_cast<const sockaddr_in6*>(ips.ipv6_->sockAddr());
-          sa6.sin6_port = htons(sip->port());
-          ipv6_source_address = std::make_shared<Network::Address::Ipv6Instance>(sa6);
-        } else {
-          ipv6_source_address = nullptr;
-        }
-      } break;
-      case Network::Address::IpVersion::v6: {
-        ipv6_source_address = src_address;
-        if (ips.ipv4_) {
-          sockaddr_in sa4 = *reinterpret_cast<const sockaddr_in*>(ips.ipv4_->sockAddr());
-          sa4.sin_port = htons(sip->port());
-          ipv4_source_address = std::make_shared<Network::Address::Ipv4Instance>(&sa4);
-        } else {
-          ipv4_source_address = nullptr;
-        }
-      } break;
-      }
+      SourceIPs sourceIPs = getL7LBEastWestSourceIPs(policy, sip, src_address);
+      ipv4_source_address = sourceIPs.ipv4;
+      ipv6_source_address = sourceIPs.ipv6;
+
```

Comment:
This would be simpler if we kept the addresses in an `IPAddressPair` type instead, then this could be just
```
sourceAddresses = getIPAddressPairFrom(sip, policy->getEndpointIPs());
```

Reply:
added an additional commit that replaces the two variables with one of type `IPAddressPair`. This way we can assign it directly.